shell/platform/darwin/macos/framework/Source/FlutterDartProject.mm

```diff
@@ -27,6 +27,10 @@ - (instancetype)initWithPrecompiledDartBundle:(NSBundle*)bundle {
   NSAssert(self, @"Super init cannot be nil");
 
   _dartBundle = bundle ?: [NSBundle bundleWithIdentifier:kAppBundleIdentifier];
+  if (_dartBundle == nil) {
```

Comment:
Could you add a comment to these new paths? It's non-obvious to someone without a lot of context why this is necessary.

Reply:
Done.